query.go

```diff
@@ -90,39 +91,48 @@ import (
 
  * *****************************************************************/
 
-/*
-QueryOne() is a convenience method that wraps Query() into a single-statement
-method.
-*/
-func (conn *Connection) QueryOne(sqlStatement string) (qr QueryResult, err error) {
+// QueryOne() is a convenience method that wraps Query() into a single-statement
+// method.
+func (conn *Connection) QueryOne(sqlStatement string, args ...interface{}) (qr QueryResult, err error) {
 	if conn.hasBeenClosed {
-		qr.Err = errClosed
-		return qr, errClosed
+		qr.Err = ErrClosed
+		return qr, ErrClosed
 	}
-	sqlStatements := make([]string, 0)
-	sqlStatements = append(sqlStatements, sqlStatement)
-	qra, err := conn.Query(sqlStatements)
+
+	qra, err := conn.Query([]string{sqlStatement}, args)
+	return qra[0], err
+}
+
+func (conn *Connection) QueryOneContext(ctx context.Context, sqlStatement string, args ...interface{}) (qr QueryResult, err error) {
+	if conn.hasBeenClosed {
+		qr.Err = ErrClosed
+		return qr, ErrClosed
+	}
+
+	qra, err := conn.QueryContext(ctx, []string{sqlStatement}, args)
 	return qra[0], err
 }
 
-/*
-Query() is used to perform SELECT operations in the database.
+// Query() is used to perform SELECT operations in the database.
+//
+// It takes an array of SQL statements and executes them in a single transaction, returning an array of QueryResult vars.
+func (conn *Connection) Query(sqlStatements []string, args ...[]interface{}) (results []QueryResult, err error) {
+	return conn.QueryContext(context.Background(), sqlStatements, args...)
+}
 
-It takes an array of SQL statements and executes them in a single transaction, returning an array of QueryResult vars.
-*/
-func (conn *Connection) Query(sqlStatements []string) (results []QueryResult, err error) {
+func (conn *Connection) QueryContext(ctx context.Context, sqlStatements []string, args ...[]interface{}) (results []QueryResult, err error) {
```

Comment:
This function expects an array of sqlStatements, but in your PR description, you have the following example:
```
rows, err := conn.QueryContext(ctx, "SELECT name FROM table WHERE id = ?", 12)
```
I guess I can't see how that would work since a `string` and not a `[]string` is being passed in the example. The same applies to other functions as well (e.g. `Query`, `Write`, and `WriteContext`). I also don't see that covered in the unit tests.

It could be the case that the PR description is not up-to-date, which is fine, but it would be great to have the official README updated with accurate usage instructions.

Reply:
Ah, thanks for the catch. I don't remember why I made it that way, but I'll take a look later this weekend.